rabbitmq-native/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java

```diff
@@ -91,19 +92,19 @@ public static Connection createConnection(BMap<BString, Object> connectionConfig
             }
             Object timeout = connectionConfig.get(RabbitMQConstants.RABBITMQ_CONNECTION_TIMEOUT);
             if (timeout != null) {
-                connectionFactory.setConnectionTimeout(Integer.parseInt(timeout.toString()));
+                connectionFactory.setConnectionTimeout((int) ((BDecimal) timeout).intValue());
```

Comment:
Is this casting correct? What happens when the value is something like `1.5`?

Reply:
The java API `connectionFactory.setConnectionTimeout()` takes only an int value. :/ 
`public void setConnectionTimeout(int timeout)`

Reply:
Is this `timeout` in `setConnectionTimeout(int timeout)` in seconds or milliseconds? In any case, I guess we have to have a conversion logic from decimal to int, instead of directly casting.

Reply:
83066c14145166b7c744da1de9ee5cc33b9b1945